tests/tests_geomstats/test_product_manifold.py

```diff
@@ -9,277 +11,286 @@
+
+space_matrix = ProductManifold(
+    manifolds=[Hypersphere(dim=2), Hyperboloid(dim=2)],
+    default_point_type="matrix",
+)
+space_vector = ProductManifold(
+    manifolds=[Hypersphere(dim=2), Hyperboloid(dim=3)], default_point_type="vector"
+)
+
+manifolds = [space_matrix, space_vector]
+
+SO3_2 = NFoldManifold(SpecialOrthogonal(3), 2)
+
+n_samples = 4
+SO3_2_point = gs.stack([gs.eye(3)] * SO3_2.n_copies * n_samples)
+SO3_2_point = gs.reshape(SO3_2_point, (n_samples, *SO3_2.shape))
+SO3_2_tangent_vec = SO3_2.to_tangent(gs.zeros((n_samples, *SO3_2.shape)), SO3_2_point)
+
+
+class TestProductManifold(TestCase, metaclass=Parametrizer):
+    cls = ProductManifold
+
+    class TestDataProductManifold(TestData):
+        def dimension_data(self):
+
+            smoke_data = [
+                dict(
+                    manifold=space_matrix,
+                    expected=4,
+                ),
+                dict(
+                    manifold=space_vector,
+                    expected=5,
+                ),
+            ]
+            return self.generate_tests(smoke_data)
+
+        def random_and_belongs_data(self):
+            n_samples = [1, 5, 10]
+            smoke_data = [
+                dict(manifold=manifold, n_samples=n_samples)
+                for (manifold, n_samples) in itertools.product(manifolds, n_samples)
+            ]
+            return self.generate_tests(smoke_data)
+
+        def exp_log_data(self):
+            n_samples = [5, 10]
+            smoke_data = [
+                dict(manifold=manifold, n_samples=n_samples)
+                for (manifold, n_samples) in itertools.product(manifolds, n_samples)
+            ]
+            return self.generate_tests(smoke_data)
+
+        def dist_log_exp_norm_data(self):
+            n_samples = [1, 5, 10]
+            smoke_data = [
+                dict(manifold=manifold, n_samples=n_samples)
+                for (manifold, n_samples) in itertools.product(
+                    [manifolds[0]], n_samples
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def inner_product_data(self):
+            smoke_data = []
+            if not tf_backend():
+                euclidean = Euclidean(3)
+                minkowski = Minkowski(3)
+                space = ProductManifold(
+                    manifolds=[euclidean, minkowski], default_point_type="matrix"
+                )
+                point = space.random_point(1)
+                expected = gs.eye(6)
+                expected[3, 3] = -1
+                smoke_data = [dict(manifold=space, point=point, expected=expected)]
+            return self.generate_tests(smoke_data)
+
+        def to_tangent_is_tangent_data(self):
+            smoke_data = [dict(manifold=space_matrix), dict(manifold=space_vector)]
+            return self.generate_tests(smoke_data)
+
+        def projection_and_belongs_vector_data(self):
```

Comment:
This should just be called: `projection_and_belongs`:
- the `_vector` was coming from the fact that the function was only tested on the default point type vector,
- but now, we are testing it on both.